### DISCUSSION TITLE
Repeating join requests without delay when no gateway is available

### DISCUSSION THREAD

**Discussion by Renger-Baulds** [Lora-net/LoRaMac-node#1397] [Q&A] [answered]
## Situation
We run src\apps\LoRaMac\fuota-test-01\SKiM881AXL\main.c on a iM881A-XL on our own board.  
Our board initialization is custom, while the application and stack are mainly a copy of the LoRaMac-node master branch of 2022-12-07.  
For our board we disable the LEDs in the example and for our test case we:  
    - provide hard-coded join EUI and app key in `LmHandlerJoinRequest()`
    - make sure no gateway is available to respond to join requests  
We use over the air activation.

## Test result
As expected, a join procedure is started from `main()` -> `LmHandlerJoin()`, after bootup and initialization:
```
###### =========== MLME-Request ============ ######
######               MLME_JOIN               ######
###### ===================================== ######
STATUS      : OK
```

As expected, the join procedure fails:
```
###### =========== MLME-Confirm ============ ######
STATUS      : Rx 2 timeout
```

### Issue
Immediately after reporting the timeout, the device starts a new join procedure, instead of entering low power mode for some seconds or minutes.  
After the expected Rx 2 timeout, the device keeps repeating this process, i.e. it sends a join request every 8 seconds.  

We figured out that `OnJoinRequest()` in main.c initiates the unexpected join request, by calling `LmHandlerJoin()` in case of an error status.  
`OnJoinRequest()` itself is called after the Rx2 timeout, as expected.  

Shouldn't `OnJoinRequest()` schedule the next join request, instead of directly starting it?

**Comment by mluis1** [comment, marked as answer]
The project provides an usage example. It is up to the final developer to modify what is provided according to its requirements.

For our example we decided to have the described behavior to speedup the evaluation process.

The only advice we can provide is for you to adapt the example code according to your needs.

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.

**Reply by Renger-Baulds** [reply]
Thanks for the clarification. I understand your decision for the example application.

However, I think it will be helpful to make developers aware of this behavior, in a line of comment above the `LmHandlerJoin()` call in `OnJoinRequest()`.

We based our application on the example and didn't realize this until many of our battery-powered devices lost their battery capacity while in stock at our customer's stores, without gateway.
